### DISCUSSION TITLE
A Few Ideas on Shortcuts

### DISCUSSION THREAD

**Discussion by LeonCheung** [SubtitleEdit/subtitleedit#9149] [Ideas]
For certain types of shortcuts, it would be nice to have some additional options to serve a concrete goal better.

E.g. for Textbox and/or list view-related shortcuts, especially for high-quality transcripts, all we need to do is just break/split them into paragraphs. In that case, we can certainly play the video automatically after the shortcuts are done. Otherwise, we have to always hit pause and play per paragraph. Normally, there could be **thousands of extra clicks** like that for a one-hour video on average. Just imagine how much time and clicks can be saved.

So, maybe it could be something like this:

![image](https://github.com/user-attachments/assets/35517f57-5762-4149-8cd2-a9eb08d0edd7)

Hopefully, we don't have to implement a bunch of potential "...and play" shortcuts then.

For some other potential shortcuts, like "Split at 1st/2nd/3rd char found" (they are very nice to have if possible), values could be defined too:

![image](https://github.com/user-attachments/assets/e00b6212-afc7-409c-8bb4-ed7f132e328a)

Or, another case:

![image](https://github.com/user-attachments/assets/5cb58046-0c62-40b9-9329-631d790bbf8e)

Or:
![image](https://github.com/user-attachments/assets/22cbe9e2-8ace-4429-a9c8-1f7a0ee2c11b)

Just listed a few. There could be many more possibilities to play with in this way. :)

**Comment by Crisses** [comment]
I have a request for the ability to customize shortcuts to do Title Case (Initial Caps), lowercase and UPPERCASE of currently selected word(s). 

Thought I'd piggyback on this request so they might be added at the same time. I like your suggestions, having to click again on the timeline to continue playing is a bit much.  You're right that space for play/pause doesn't work in the edit field.

Just discovered aAn alternative is to have a different key (i.e. F-something) to be play/pause.  I just changed it to F10 (under Options -> Shortcuts…).

I use a Contour ShuttlePROv2 and custom program it for the app I'm using, so it doesn't matter to the Shuttle what my "play/pause" button is, so long as it toggles playing. I changed the thumb key to send F10 to the computer instead of space. I put space on another key for convenience sake while editing text (I have period, space, return and delete on my Shuttle in addition to other keys). So right hand on the mouse, left hand on the shuttle… 

If you use macros (like in Keyboard Maestro or any other macro app/editor) then you can add the play key after the shortcut for say merging or splitting lines (now I'm wondering if I can macro the Title Case thing lol).
